File: client/Research2DPrototype/Assets/Scripts/SI/BaseUnits.cs
using System;

public class BaseUnits
{

  /// <summary>
  /// # s^p
  /// </summary>
  /// <param name="p">power as int</param>
  /// <returns></returns>
  protected static SiUnit Second(int p)
  {
    return new SiUnit(p, 0, 0, 0, 0, 0, 0);
  }
  /// <summary>
  /// # m^p
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Meter(int p)
  {
    return new SiUnit(0, p, 0, 0, 0, 0, 0);
  }
  /// <summary>
  /// # kg^p
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Kilogram(int p)
  {
    return new SiUnit(0, 0, p, 0, 0, 0, 0);
  }
  /// <summary>
  /// # A^p -> https://en.wikipedia.org/wiki/Ampere
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Ampere(int p)
  {
    return new SiUnit(0, 0, 0, p, 0, 0, 0);
  }
  /// <summary>
  /// # K^p
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Kelvin(int p)
  {
    return new SiUnit(0, 0, 0, 0, p, 0, 0);
  }
  /// <summary>
  /// # mol^p
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Mole(int p)
  {
    return new SiUnit(0, 0, 0, 0, 0, p, 0);
  }
  /// <summary>
  /// # cd^p
  /// </summary>
  /// <param name="p"></param>
  /// <returns></returns>
  protected static SiUnit Candela(int p)
  {
    return new SiUnit(0, 0, 0, 0, 0, 0, p);
  }
}

File: shared_data/base_units.rb
require_relative 'si_unit'

# derived units return si unit powers
class BaseUnits
    # s^p
    def Second(p)
        return SiUnit.new(p, 0, 0, 0, 0, 0, 0)
    end
    # m^p
    def Meter(p)
        return SiUnit.new(0, p, 0, 0, 0, 0, 0)
    end
    # kg^p
    def Kilogram(p)
        return SiUnit.new(0, 0, p, 0, 0, 0, 0)
    end
    # A^p -> https://en.wikipedia.org/wiki/Ampere
    def Ampere(p)
        return SiUnit.new(0, 0, 0, p, 0, 0, 0)
    end
    # K^p
    def Kelvin(p)
        return SiUnit.new(0, 0, 0, 0, p, 0, 0)
    end
    # mol^p
    def Mole(p)
        return SiUnit.new(0, 0, 0, 0, 0, p, 0)
    end
    # cd^p
    def Candela(p)
        return SiUnit.new(0, 0, 0, 0, 0, 0, p)
    end
end
